### DISCUSSION TITLE
Ability to add default tracker list

### DISCUSSION THREAD

**Discussion by ragebflame** [haugene/docker-transmission-openvpn#2256] [Ideas]
Hi,

From looking at other torrent clients, it would be useful to auto add a list of trackers to every new torrent. I find this helps get stuck torrents over the line, but there may be downsides I'm unaware of. From searching online I'm not even sure that Transmission itself supports this.

For example, the feature in qBittorrent:
_In qBittorrent, go to Tools, Preferences, BitTorrent, select "Automatically add these trackers to new downloads" and paste in the list. Click OK._

There are good public tracker lists maintained and available on Github. e.g. [https://ngosang.github.io/trackerslist/](https://ngosang.github.io/trackerslist/)

There look to be some bash script options on the go also:

- https://github.com/AndrewMarchukov/tracker-add
- https://github.com/oilervoss/transmission
- https://github.com/Jorman/Scripts#addtransmissiontrackerssh

**Comment by pkishino** [comment]
Yeah, check here I believe for details
https://github.com/transmission/transmission/issues/667
there might be patch or build available.. but again, that would be on transmission end and nothing on the container

**Reply by ragebflame** [reply]
Thanks. I have a working solution using the Transmission environment variables and one of the [bash scripts](https://github.com/Jorman/Scripts#addtransmissiontrackerssh) listed above.

```text
      - TRANSMISSION_SCRIPT_TORRENT_ADDED_ENABLED=true
      - TRANSMISSION_SCRIPT_TORRENT_ADDED_FILENAME="/data/transmission-home/AddTransmissionTrackers.sh"
```

**Reply by pkishino** [reply]
If you want I’d be happy to approve adding a section on the wiki if you can write it up please :)

**Reply by ragebflame** [reply]
Sure. Does it suit to write it up here? Or is there a better location to draft it?

**Reply by ragebflame** [reply]
I may have spoken too soon on this actually. It does not look like the script is running, although permissions look correct.

I am getting ahead of myself here. Looks like the PR with this feature [has been merged](https://github.com/transmission/transmission/pull/1896), but is yet to be released.

**Reply by pkishino** [reply]
you can directly edit here:
https://haugene.github.io/docker-transmission-openvpn/
perhaps under the tips&tricks section

**Reply by pkishino** [reply]
ah, ok

**Reply by ragebflame** [reply]
I'll pick this back up when things are in place on the transmission side.

**Comment by Forage** [comment]
I was looking for the same behaviour, though I believe it could/should be addressed differently.
Transmission has the `Default Public Trackers` setting in its settings. Trackers added here will be automagically used for all torrents. No additional manual scripting needed.
The field can be set through the API using `default-trackers` if I'm not mistaken.

docker-transmission-openvpn could provide a new environment variable allowing you to provide one or more URLs to tracker lists. These URLs could then be downloaded and its text inserted through the API.
To make things more exotic:
- Update the lists at a fixed (daily?) interval
- Update the lists at an interval set through an environment variable
- Provide tracker URLs directly through an environment variable (instead of a URL to a list of trackers)
